packages/dom/src/node.rs

```diff
@@ -797,6 +800,50 @@ impl Node {
                 y,
             }))
     }
+
+    /// Scroll this node by given x and y
+    /// Will bubble scrolling up to parent node once it can no longer scroll further
+    pub fn scroll_by(&mut self, x: f64, y: f64) {
+        self.scroll_offset.x += x;
+        self.scroll_offset.y += y;
+
+        //If we're past our scroll bounds, transfer scrolling to parent
+        if self.scroll_offset.x < 0.0
+            || self.scroll_offset.x > self.final_layout.scroll_width() as f64
+            || self.scroll_offset.y < 0.0
+            || self.scroll_offset.y > self.final_layout.scroll_height() as f64
```

Comment:
This check should occur before updating the node's scroll offset. It should also allow *partial* transfer of the scroll delta to the parent. For example, if `scroll_offset.y` is 10 and x is `-20` then the current node should scroll by `-10` to scroll offset `0`, and the parent should be scrolled by `10`.

Reply:
I believe if you did this then you wouldn't need to clamp.